connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java

```diff
@@ -50,6 +50,7 @@
     private static final Logger log = LoggerFactory.getLogger(WorkerSourceTask.class);
 
     private static final long SEND_FAILED_BACKOFF_MS = 100;
+    private static final long UNPAUSE_AWAIT_TIMEOUT_MS = 100;
```

Comment:
Doesn't unpausing just wait indefinitely? Why shouldn't this value be effectively infinite? The code in WorkerTask is supposed to wake up the waiting thread anyway via `notifyAll`.

Relatedly, during a `WorkerTask.stop()`, shouldn't there also be a `this.notifyAll()`? Otherwise we'll unnecessarily have to continue sleeping for the duration of this unpause timeout waiting for the task to quit when we `awaitStop()`.